File: backend-api/RiftARENA/RiftARENA/Models/Tournament.cs
using RiftARENA.Models;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace RiftArena.Models
{

    public enum Status
    {
        Published, NotPublished, Canceled, Soon, Online, Closed
    }
    public class Tournament
    {
        [Key]
        public int TournamentId { get; set; }

        public int NumberOfTeams { get; set; }

        public string Name { get; set;}

        public string Description { get; set;}

        public string Rank { get; set; }

        public string Region { get; set; }

        public string FinalWinner { get; set; }

        public float Prize { get; set; }

        public string MiniumTier { get; set; }

        public string Poster { get; set; }

        public List<Team> Stages { get; set; }

        public Status State { get; set; }

        public List<Messages> chat { get; set; }

        //data

        
        









    }
}

File: angular-client-side/src/app/models/tournament.ts
import {Team} from './team';
import { Message } from './message';

export enum State {
    PUBLISHED,
    NOT_PUBLISHED,
    CANCELED,
    SOON,
    ONLINE,
    CLOSED
}

export class Tournament {
    constructor(
        public tournamentId: number,
        public numberOfTeams: number,
        public maxTeams: number,
        public name: string,
        public state: State,
        public stages: Team[],
        public rank: string,
        public date: any,
        public region: string,
        public finalWinner: string,
        public description: string,
        public prize: number,
        public poster: string,
        public chat: Message[],
    ) {}

    static fields(){
        return {
            inputs: [
              {
                name: 'name',
                type: 'text',
                placeholder: 'Enter tournament name',
                iconlabel: 'category icon',
                icon: 'category',
                model: undefined,
              },
              {
                name: 'nTeams',
                type: 'number',
                placeholder: 'Enter the number of teams',
                iconlabel: 'group icon',
                icon: 'group',
                model: undefined,
              },
              {
                name: 'date',
                type: 'datetime-local',
                placeholder: 'Enter tournament date',
                iconlabel: 'today icon',
                icon: 'today',
                model: undefined,
              },
        ]}
    }
}
